src/challenges/challengeOne/2.Dropdown/Dropdown.test.tsx

```diff
@@ -1,8 +1,33 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Dropdown } from "./Dropdown";
-
-// describe("Dropdown", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { Dropdown, DropdownProps } from "./Dropdown";
+
+describe("Dropdown", () => {
+  const mockOptions = ["Option 1", "Option 2", "Option 3"];
+  const mockLabel = "Dropdown Label";
+  const mockId = "dropdownId";
+
+  const renderComponent = (additionalProps?: Partial<DropdownProps>) => {
+    const defaultProps: Partial<DropdownProps> = {
+      options: mockOptions,
+      label: mockLabel,
+      id: mockId,
+    };
+    render(<Dropdown {...defaultProps} {...additionalProps} />);
+  };
+
+  it("renders the dropdown label", () => {
+    renderComponent();
+    expect(screen.getByLabelText(mockLabel)).toBeInTheDocument();
+  });
+
+  it("renders the dropdown options", () => {
+    renderComponent();
+    const dropdown = screen.getByRole("combobox");
+    expect(dropdown).toBeInTheDocument();
+    expect(dropdown.children.length).toBe(mockOptions.length);
+    mockOptions.forEach((option) => {
```

Comment:
This is an interesting approach but is very brittle. What if we change the structure of our dropdown to have a hidden option? Or add a design element like a divider between options? Then the number of children would increase. It's unlikely, but your test would break because it's testing the wrong thing. 

We don't care how many children dropdown has, we care what the user _sees_. 

This test would fail if you were actually testing what the user sees. Think about it, when you render the Dropdown, it's closed. It only opens and displays the options when you click on the button.

So if I were to write three tests to check that my options display

```
expect(screen.getByText("Option 1")).toBeInTheDocument();
expect(screen.getByText("Option 2")).toBeInTheDocument();
expect(screen.getByText("Option 3")).toBeInTheDocument();
```
They shouldn't pass until I have opened my dropdown. So you would need 

`await userEvent.click(dropdown);`

So a better test would be 

```
it("renders the dropdown options when the dropdown button is clicks", async () => {
   renderComponent();
   
   expect(screen.queryByText("Option 1")).not.toBeInTheDocument();
   expect(screen.queryByText("Option 2")).not.toBeInTheDocument();
   expect(screen.queryByText("Option 3")).not.toBeInTheDocument();

   await userEvent.click(screen.getByRole("combobox"));

   expect(screen.getByText("Option 1")).toBeInTheDocument();
   expect(screen.getByText("Option 2")).toBeInTheDocument();
   expect(screen.getByText("Option 3")).toBeInTheDocument();
});

As a final side note, you never want logic in your tests, so no loops or `forEach` - it's bad practice as it's very easy for your logic to fail and cause problems with your test.

You can of course use the mocks ` expect(screen.getByText(mockOptions[0])).toBeInTheDocument();` but I personality prefer to show the text so the developer knows exactly what is being displayed right there in the test.